packages/app/pages/AllProjectsPage/AllProjectsPage.tsx

```diff
@@ -19,43 +21,140 @@ interface Props {
   children: Object
 }
 
+/**
+ * Row of buttons to filter which user agent types should be displayed
+ */
+class NavBar extends React.Component {
+
+  readonly navButtons = ["All", "Cooperative", "Projects", "Organizations", "Groups"]
+
+  private currentTheme;
+
+  constructor(private props) {
+    super(props)
+    this.state = {
+      activeButton: "All",
+    };
+
+    this.currentTheme = themeable(this.props.theme)
+  }
+
+  /**
+   * Handler function for when buttons are pressed.
+   * TODO apply the filter to the user agents
+   * @param buttonName the text of the button that was clicked
+   */
+  handleClick = (buttonName) => {
+    this.setState({ activeButton: buttonName })
+    this.forceUpdate(() => console.log('Finished Updating'))
+    // this.state.activeButton  = buttonName
+    console.log("Clicked", buttonName)
+    // TODO refresh the list and filter
```

Comment:
Just a question: What does this mean?

Reply:
What does "this" mean? 😛

Reply:
The TODO

Reply:
The current projects page has buttons on it to filter by different types of user agents ("All", "Cooperative", "Projects", "Organizations", "Groups"). The buttons were not functional (they were just text) at the start of the spike. As part of the spike they were turned into buttons and an action listener was attached, but the filtering functionality wasn't implemented yet. It has not been in scope of a sprint since, so the TODO just marks that the action still needs to happen but it hasn't been scheduled.